exn.ml

```diff
@@ -33,7 +33,14 @@ let to_string exn =
 let str = to_string
 
 let fail ?exn fmt =
-  let fails s = match exn with None -> failwith s | Some exn -> failwith (s ^ " : " ^ to_string exn) in
+  let fails s =
+    match exn with
+    | None -> failwith s
+    | Some original_exn ->
+      let orig_bt = Printexc.get_raw_backtrace () in
```

Comment:
is it correct? What if `exn` wasn't the last exn raised?

Reply:
you are right, it is not hard to come up with the case when it is not correct, thought i would argue most of real-world usecases should be ok. Did you see a specific case when this was wrong?

Reply:
nothing specific. I know this is something we have faced over the years. But not particularly since this change landed.

Reply:
but yes, it deserves a comment, @tatchi please

Reply:
https://github.com/ahrefs/devkit/pull/31